schema/datamodel.py

```diff
@@ -1,43 +1,52 @@
 import xml.etree.ElementTree as ET
-from collections import namedtuple
+
+from collections import OrderedDict
+from itertools import tee, chain
 from pathlib import Path
-import os
 
-precoRevenda = namedtuple('PrecoRevenda', 'valor_compra valor_venda unidade municipio regiao estado bandeira produto razao_social_revenda data_coleta')
 
-class PrecosParser():
+TAG_PREFIX = '{urn:schemas-microsoft-com:xml-analysis:rowset}'
+
+TAG_KEY_MAPPING = OrderedDict({
+    "0": "valor_compra",
+    "1": "valor_venda",
+    "2": "unidade",
+    "3": "municipio",
+    "4": "regiao",
+    "5": "estado",
+    "6": "bandeira",
+    "7": "produto",
+    "8": "razao_social_revenda",
+    "9": "data_coleta",
+})
+
+def print_preco(obj=None, sep=";"):
+    if not obj:
+        obj = TAG_KEY_MAPPING
+
+    txt = sep.join(obj.values())
```

Comment:
Como PrecoRevenda é iniciado como float('nan') e nos casos em que `valor_compra` for None haverá um type error:

```
TypeError                                 Traceback (most recent call last)
~\Documents\datascience\anp-dados-abertos\parser_2.py in <module>()
     21     #exporta para um arquivo csv
     22     filename = file_.name[:-4] #coleta o ano do dado
---> 23     filedata.to_csv(f"{filename}.csv", filedata.data_container)
     24
     25     #adiciona os dados da instância ao container da classe

~\Documents\datascience\anp-dados-abertos\schema\datamodel_2.py in to_csv(nome_arquivo, all_data)
    113
    114             for preco in data_list:
--> 115                 f.write(print_preco(preco))
    116
    117         print(f"Arquivo {fullpath} criado com sucesso.")

~\Documents\datascience\anp-dados-abertos\schema\datamodel_2.py in print_preco(obj, sep)
     25         obj = TAG_KEY_MAPPING
     26
---> 27     txt = sep.join(obj.values())
     28
     29     return f"{txt}\n"

TypeError: sequence item 0: expected str instance, float found
```

Portanto, é necessário transformar garantir que o método seja aplicado sobre strings.

```
def print_preco(obj=None, sep=";"):
    if not obj:
        obj = TAG_KEY_MAPPING

    txt = sep.join(str(obj.values()))

    return f"{txt}\n"
```

Reply:
Verdade. Deveria ser

`txt = sep.join(map(str, obj.values()))`